miro/provider/client.py

```diff
@@ -0,0 +1,54 @@
+import requests
+from flask import current_app as app, request
+
+from . import UpstreamProviderError
+
+AUTHORIZATION_HEADER = "Authorization"
+BEARER_PREFIX = "Bearer "
+DEFAULT_LIMIT = 20
+
+
+class MiroClient:
+    BASE_PATH = "https://api.miro.com/v2"
+    BOARDS_ENDPOINT = "/boards"
+
+    def __init__(self, access_token, limit):
+        self.headers = {
+            "Authorization": f"Bearer {access_token}",
+        }
+        self.limit = limit
+
+    def search(self, query):
+        url = self.BASE_PATH + self.BOARDS_ENDPOINT
+        params = {"query": query, "limit": self.limit}
+        response = requests.get(
+            url,
+            headers=self.headers,
+            params=params,
+        )
+        if response.status_code != 200:
+            message = response.text or f"Error: HTTP {response.status_code}"
+            raise UpstreamProviderError(message)
+
+        results = response.json()["data"]
+
+        return results
+
+
+def get_client():
+    access_token = app.config.get("ACCESS_TOKEN", None)
```

Comment:
You can save lines 41-42 by changing this to `app.config.get("ACCESS_TOKEN", get_access_token())`

Reply:
Good point